lib/internal/modules/esm/translators.js

```diff
@@ -193,8 +201,10 @@ function enrichCJSError(err, content, filename) {
     // a SyntaxError that will throw and likely terminate the process before an
     // asynchronous warning would be emitted.
     emitWarningSync(
-      'To load an ES module, set "type": "module" in the package.json or use ' +
-      'the .mjs extension.',
+      'To load an ES module:\n' +
+      '- Either the nearest package.json should set "type":"module", ' +
+      'or the module should use the .mjs extension.\n' +
+      '- If it\'s loaded using require(), use --experimental-require-module.',
```

Comment:
There’s already an open PR about adding a mention of `--experimental-detect-module` here. I feel like our error messages shouldn’t recommend experimental features; though if we do, it should mention both flags.

Reply:
This only shows up in `require()`, which `--experimental-detect-module` doesn't work with for now. I think it should only mention it when they work together.

Reply:
I just don’t think we should be mentioning experimental flags as recommendations, in general. Arguably we should be mentioning `--experimental-default-type`, too, but I think its days are numbered. https://github.com/nodejs/node/pull/51963#issuecomment-1977746234

I think for now just leave the error message alone? We don’t need to add mentions of various experimental flags. We should only recommend fixes that are stable.

Reply:
I'm fine with leaving it out of this PR and open a follow-up to add it into the message. Personally I think it's okay or even good to mention experimental features in error messages, so that we get more people to learn about them and send feedback to us, helping us making the feature mature, especially when it comes to ERR_REQUIRE_ESM which has been frustrating for years. I don't think it's the same as `--experimental-detect-module`, even - to fix ambiguous module types, the fix should be what's already mentioned in the error message - use .mjs or type field, not using detection. Detection is not a fix, and should be avoided instead of being recommended.

Reply:
Let’s leave it out of this PR if that’s okay, and discuss it within https://github.com/nodejs/node/pull/51963? We can hold that PR to land after this one.